src/gflownet/models/bengio2021flow.py

```diff
@@ -0,0 +1,279 @@
+        self.num_conv_steps = num_conv_steps
+        self.dropout_rate = 0
+
+        self.act = nn.LeakyReLU()
+
+        net = nn.Sequential(nn.Linear(4, 128), self.act, nn.Linear(128, dim * dim))
+        self.conv = NNConv(dim, dim, net, aggr='mean')
+        self.gru = nn.GRU(dim, dim)
+
+        self.lin1 = nn.Linear(dim, dim * 8)
+        self.lin2 = nn.Linear(dim * 8, num_out_per_stem)
+
+        self.set2set = Set2Set(dim, processing_steps=3)
+        self.lin3 = nn.Linear(dim * 2, num_out_per_mol)
+        self.bond2out = nn.Sequential(nn.Linear(dim * 2, dim), self.act, nn.Linear(dim, dim), self.act,
+                                      nn.Linear(dim, num_out_per_bond))
+
+    def forward(self, data, vec_data=None, do_stems=True, do_bonds=False, k=None, do_dropout=False):
+        out = self.act(self.lin0(data.x))
+        h = out.unsqueeze(0)
+        h = F.dropout(h, training=do_dropout, p=self.dropout_rate)
+
+        for i in range(self.num_conv_steps):
+            m = self.act(self.conv(out, data.edge_index, data.edge_attr))
+            m = F.dropout(m, training=do_dropout, p=self.dropout_rate)
+            out, h = self.gru(m.unsqueeze(0).contiguous(), h.contiguous())
+            h = F.dropout(h, training=do_dropout, p=self.dropout_rate)
+            out = out.squeeze(0)
+
+        global_out = self.set2set(out, data.batch)
+        global_out = F.dropout(global_out, training=do_dropout, p=self.dropout_rate)
+        per_mol_out = self.lin3(global_out)  # per mol scalar outputs
+        return per_mol_out
+
+
+def load_original_model():
+    num_feat = (14 + 1 + len(atomic_numbers))
+    mpnn = MPNNet(num_feat=num_feat, num_vec=0, dim=64, num_out_per_mol=1, num_out_per_stem=105, num_conv_steps=12)
+    f = requests.get("https://github.com/GFNOrg/gflownet/raw/master/mols/data/pretrained_proxy/best_params.pkl.gz",
+                     stream=True)
+    params = pickle.load(gzip.open(f.raw))  # nosec
+    param_map = {
+        'lin0.weight': params[0],
+        'lin0.bias': params[1],
+        'conv.bias': params[3],
+        'conv.nn.0.weight': params[4],
+        'conv.nn.0.bias': params[5],
+        'conv.nn.2.weight': params[6],
+        'conv.nn.2.bias': params[7],
+        'conv.lin.weight': params[2],
+        'gru.weight_ih_l0': params[8],
+        'gru.weight_hh_l0': params[9],
+        'gru.bias_ih_l0': params[10],
+        'gru.bias_hh_l0': params[11],
+        'lin1.weight': params[12],
+        'lin1.bias': params[13],
+        'lin2.weight': params[14],
+        'lin2.bias': params[15],
+        'set2set.lstm.weight_ih_l0': params[16],
+        'set2set.lstm.weight_hh_l0': params[17],
+        'set2set.lstm.bias_ih_l0': params[18],
+        'set2set.lstm.bias_hh_l0': params[19],
+        'lin3.weight': params[20],
+        'lin3.bias': params[21],
+    }
+    for k, v in param_map.items():
+        mpnn.get_parameter(k).data = torch.tensor(v)
+    return mpnn
+
+
+atomic_numbers = {
```

Comment:
Is this full dictionary useful (only it's `len` is used)? If so would it make sense to add it to some Constants.py file or if not then just move to the top of this file for improved readability?

Reply:
True, I'll add a constant at the top of the file.